libraries/bot-azure/src/main/java/com/microsoft/bot/azure/queues/AzureQueueStorage.java

```diff
@@ -0,0 +1,91 @@
+     * jsonSerializer.TypeNameHandling = TypeNameHandling.None.
+     * jsonSerializer.NullValueHandling = NullValueHandling.Ignore.
+     * @param queuesStorageConnectionString Azure Storage connection string.
+     * @param queueName Name of the storage queue where entities will be queued.
+     */
+    public AzureQueueStorage(String queuesStorageConnectionString, String queueName) {
+        if (StringUtils.isBlank(queuesStorageConnectionString)) {
+            throw new IllegalArgumentException("queuesStorageConnectionString is required.");
+        }
+
+        if (StringUtils.isBlank(queueName)) {
+            throw new IllegalArgumentException("queueName is required.");
+        }
+
+        _queueClient = new QueueClientBuilder()
+            .connectionString(queuesStorageConnectionString)
+            .queueName(queueName)
+            .buildClient();
+    }
+
+    /**
+     * Queue an Activity to an Azure.Storage.Queues.QueueClient. The visibility timeout specifies how long the message should be invisible
+     * to Dequeue and Peek operations. The message content must be a UTF-8 encoded string that is up to 64KB in size.
+     * @param activity This is expected to be an @see Activity retrieved from a call to
+     *                 activity.GetConversationReference().GetContinuationActivity(). This enables restarting the conversation
+     *                 using BotAdapter.ContinueConversationAsync.
+     * @param visibilityTimeout Default value of 0. Cannot be larger than 7 days.
+     * @param timeToLive Specifies the time-to-live interval for the message.
+     * @return @see SendReceipt as a Json string, from the QueueClient SendMessageAsync operation.
+     */
+    @Override
+    public CompletableFuture<String> QueueActivity(Activity activity, @Nullable Duration visibilityTimeout, @Nullable Duration timeToLive) {
+        return CompletableFuture.supplyAsync(() -> {
+            if (_createQueueIfNotExists) {
+                // This is an optimization flag to check if the container creation call has been made.
+                // It is okay if this is called more than once.
+                _createQueueIfNotExists = false;
+                try {
+                    _queueClient.create();
+                } catch (QueueStorageException e) {
+
+                }
+            }
+
+            try {
+                ObjectMapper objectMapper = new ObjectMapper();
+                String serializedActivity = objectMapper.writeValueAsString(activity);
+                byte[] encodedBytes = serializedActivity.getBytes(StandardCharsets.UTF_8);
+                String encodedString = Base64.getEncoder().encodeToString(encodedBytes);
+
+                SendMessageResult receipt = _queueClient.sendMessage(encodedString);
+                return objectMapper.writeValueAsString(receipt);
+            } catch (JsonProcessingException e) {
+                e.printStackTrace();
+            }
+            return null;
+        });
```

Comment:
I'm not quite sure about this, can be simplified as:
```suggestion
        if (createQueueIfNotExists) {
                // This is an optimization flag to check if the container creation call has been made.
                // It is okay if this is called more than once.
                queueClient.create();
        }
        
        JacksonAdapter jacksonAdapter = new JacksonAdapter();
        String message = jacksonAdapter.serialize(activity);
        SendMessageResult receipt = queueClient.sendMessage(message);
        
        return CompletableFuture.completedFuture(jacksonAdapter.serialize(receipt.getValue()));
```

Also, we can have more parity, what do you think?

Reply:
The `try` bock is required. In java several methods call require to be inside a try block because otherwise it is treated as a unhandled exception. In this case it is `objectMapper.writeValueAsString(receipt)` to be inside the `try catch` block.

Similar for the `queueClient.create();`. Java does not have TryToCreate method, but has `create` [that will throw an exception if it already exists](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClient.java#L103). So I surrounded it with a try catch to discard **_only_** the QueueStorageException exception. This is similar to how it was created in [CosmosDbPartitionedStorage](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-azure/src/main/java/com/microsoft/bot/azure/CosmosDbPartitionedStorage.java#L279-L290).